filip/clients/ngsi_v2/cb.py

```diff
@@ -880,18 +881,39 @@ def get_subscription_list(self,
             self.log_error(err=err, msg=msg)
             raise
 
-    def post_subscription(self, subscription: Subscription) -> str:
+    def post_subscription(self, subscription: Subscription,
+                          update: Optional[bool] = False) -> str:
         """
         Creates a new subscription. The subscription is represented by a
         Subscription object defined in filip.cb.models.
 
-        Args:
-            subscription:
+        If the subscription already exists, the adding is prevented and the id
+        of the existing subscription is returned.
 
+        Args:
+            subscription: Subscription
+            update: True - If the subscription already exists, update it
+                    False- If the subscription already exists, throw warning
         Returns:
-            object: 
+            str: Id of the (created) subscription
 
         """
+        existing_subscriptions = self.get_subscription_list()
+
+        sub_hash = str(subscription.dict(include={'subject', 'notification'}))
+        for ex_sub in existing_subscriptions:
+            if sub_hash ==\
+                    str(ex_sub.dict(include={'subject', 'notification'})):
```

Comment:
If you cast it to a str, why don't you use to_json(include:{})?

Reply:
I needed to cast it to str as he would otherwise compare the object pointers. I did not found a better suited methode to achieve this result. I can not use to_json() as subscription is a pydantic model and not a pandas dataframe.